include/boost/compute/context.hpp

```diff
@@ -175,27 +175,13 @@ class context
     /// devices, the first is returned.
     device get_device() const
     {
-        size_t count = 0;
-        clGetContextInfo(m_context,
-                         CL_CONTEXT_DEVICES,
-                         0,
-                         0,
-                         &count);
-        if(count == 0){
-            return device();
-        }
+        std::vector<device> devices = get_devices();
 
-        cl_device_id id;
-        clGetContextInfo(m_context,
-                         CL_CONTEXT_DEVICES,
-                         sizeof(cl_device_id),
-                         &id,
-                         0);
-        if(id == 0){
+        if(devices.empty()) {
             return device();
         }
 
-        return device(id);
+        return devices.front();
```

Comment:
You tried it also with creating one context with both devices? So first call to clGetContextInfo returns more than 8 byte as result?
For me this fix works. Maybe the function is only able to return all devices and not only one?